File: leetcode代码/递归+回溯/24. 两两交换链表.dart
Node? swap(Node? head) {
  if (head == null || head.next == null) {
    return head;
  }
  final ans = head.next;
  dfs(null, head, head.next, head.next?.next);
  return ans;
}

void dfs(Node? last, Node? left, Node? right, Node? next) {
  if (left == null || right == null) {
    return;
  }
  if (last != null) {
    last.next = right;
  }
  right.next = left;
  left.next = next;
  dfs(
    left,
    left.next,
    left.next?.next,
    left.next?.next?.next,
  );
}

File: leetcode代码/递归+回溯/24. 两两交换链表.swift
class Node {
    var val: Int = 0
    var next: Node?
}

func swapLinkedList(head: Node?) -> Node? {
    if (head == nil || head!.next == nil) {
        return head
    }
    let ans = head?.next
    dfs(last: nil, left: head, right: head?.next, next: head?.next?.next)
    return ans
}

func dfs(last: Node?, left: Node?, right: Node?, next: Node?) -> Void {
    if (left == nil || right == nil) {
        return
    }
    if (last != nil) {
        last?.next = right
    }
    right?.next = left
    left?.next = next
    dfs(last: left, left: left?.next, right: left?.next?.next, next: left?.next?.next?.next)
}